presto-main/src/main/java/com/facebook/presto/sql/planner/SortExpressionExtractor.java

```diff
@@ -57,14 +63,15 @@ by sorting position links according to the result of f(...) function.
      */
     private SortExpressionExtractor() {}
 
-    public static Optional<SortExpressionContext> extractSortExpression(Set<Symbol> buildSymbols, Expression filter)
+    public static Optional<SortExpressionContext> extractSortExpression(Set<Symbol> buildSymbols, RowExpression filter, FunctionManager functionManager)
     {
-        List<Expression> filterConjuncts = ExpressionUtils.extractConjuncts(filter);
-        SortExpressionVisitor visitor = new SortExpressionVisitor(buildSymbols);
+        List<RowExpression> filterConjuncts = LogicalRowExpressions.extractConjuncts(filter);
+        SortExpressionVisitor visitor = new SortExpressionVisitor(buildSymbols, functionManager);
 
+        com.facebook.presto.sql.relational.DeterminismEvaluator determinismEvaluator = new com.facebook.presto.sql.relational.DeterminismEvaluator(functionManager);
```

Comment:
nit: can we rename one of them so that we don't have to use a full class path?

Reply:
@mbasmanova has this change in https://github.com/prestodb/presto/pull/12729/commits/62dc9fe4d6850b15ce45d37ad85b6e863a5a2965. So I will just leave the change to her.